examples/example_plugin/README.md

```diff
@@ -0,0 +1,21 @@
+# gooseBit Example Plugin
+
+This is a basic example plugin to provide a framework for creating custom plugins for gooseBit.
+
+## Plugin Requirements
+
+Plugins are identified automatically with entry points. This means that you must define the relevant section in your `pyproject.toml` file.
+
```

Comment:
I'm thinking we should define plugins in the config file and dynamically load them based off of the import path in the config file.

Reply:
The standard for this seems to be just that you can install them and they will automatically get loaded.  As long as  users are properly keeping track of dependencies this wont be a problem.

You can test this by running `poetry install --with examples` to load the plugin, and `poetry install --without examples` to unload it.

Reply:
> The standard for this seems to be just that you can install them and they will automatically get loaded.

I'm not that big a fan of auto-loading just based on what's installed, IMO they should be explicitly enabled in a config file as well so that merely installing plugins to the python environment doesn't actually change the behavior of the application(which would be somewhat unexpected behavior).

> As long as users are properly keeping track of dependencies this wont be a problem.

Yeah...so I don't think we want to change behavior based on installed deps in this way, I mean if a user installs a plugin but then decides not to use it they should be able to just not have it configured for it to not be loaded or something like that.

Reply:
Ok, so we need another option in settings for `plugins`, which can be a list of plugin names.  Shouldn't be too hard.

Reply:
I think each plugin should be a configurable class(which can import various hooks) and the plugin name should be the import path.

Reply:
And configure like this maybe:
```yaml
plugins:
  upstreamdata.cfd.provisioner:
    configkey1: configval1
    configkey2: configval2
    configkey3:
      subkey1: subval1
      subkey2: subval2
```